lib/Runtime/Library/WebAssemblyInstance.cpp

```diff
@@ -328,9 +328,14 @@ void WebAssemblyInstance::LoadImports(
                     JavascriptError::ThrowWebAssemblyLinkError(ctx, WASMERR_NeedMemoryObject);
                 }
                 WebAssemblyMemory * mem = WebAssemblyMemory::FromVar(prop);
-                if (!wasmModule->IsValidMemoryImport(mem))
+
+                if (mem->GetInitialLength() < wasmModule->GetMemoryInitSize())
                 {
-                    JavascriptError::ThrowWebAssemblyLinkError(ctx, WASMERR_NeedMemoryObject);
+                    throw Wasm::WasmCompilationException(_u("Imported memory initial size (%u) is smaller than declared (%u)"), mem->GetInitialLength(), wasmModule->GetMemoryInitSize());
```

Comment:
`CreateInstance` has a try/catch for Compilation exception and converts them into LinkError, This allows more control over the error message.
I don't think we can avoid the try/catch as a lot of our module methods can throw.
The alternative would be to create another entry in rterrors.h, but since this particular error can really only happen here, I felt this made more sense